src/smc-webapp/course/main.cjsx

```diff
@@ -1549,20 +1558,30 @@ init_redux = (course_filename, redux, course_project_id) ->
         get_students: =>
             @get('students')
 
+        # Uses an instructor given name if it exists
         get_student_name: (student, include_email=false) =>
             student = @get_student(student)
             if not student?
                 return 'student'
             email = student.get('email_address')
-            name = user_store.get_name(student.get('account_id'))
-            n = name ? email ? 'student'
+            account_id = student.get('account_id')
+            first_name = student.get('first_name') ? user_store.get_first_name(account_id)
```

Comment:
@timothyclemansinsea see the line above, there is something going on how the first and last names of student's are derived.